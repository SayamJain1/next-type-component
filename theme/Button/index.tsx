type ButtonProps = {
  text: string;
  rightIcon?: React.ReactNode;
  leftIcon?: React.ReactNode;
  className?: string;
  disable?: boolean;
  type?: "submit" | "button" | "reset";
  onClick?: () => void;
  variant: "primary-btn" | "secondary-btn" | "text-blue-btn";
  size: "sm" | "md" | "lg";
  gap?: string;
};

const Button: React.FC<ButtonProps> = ({
  text,
  rightIcon,
  leftIcon,
  className,
  disable,
  type,
  onClick,
  variant,
  size,
  gap,
}) => {
  return (
    <button
      type={type}
      className={`inline-block w-full ${variant} ${size} rounded-lg  ${
        disable ? "opacity-50 pointer-events-none" : ""
      } ${className}`}
      disabled={disable}
      onClick={onClick}
    >
      <span className={`flex items-center justify-center ${gap}`}>
        {leftIcon ? leftIcon : ""}
        {text}
        {rightIcon ? rightIcon : ""}
      </span>
    </button>
  );
};
export default Button;

// ${
//   !disable && "hover:opacity-90"
// }
