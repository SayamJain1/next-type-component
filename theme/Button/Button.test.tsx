import React from "react";
import renderer from "react-test-renderer";
import Button from "@/theme/Button/index";

// import Button from "./Button";

it("performs snapshot testing", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
