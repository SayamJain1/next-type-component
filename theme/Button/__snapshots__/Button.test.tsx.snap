// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs snapshot testing 1`] = `
<button
  className="flex items-center undefined undefined rounded-lg undefined undefined"
/>
`;
